### ISSUE TITLE
Unable to declare ImageField as required

### ISSUE THREAD

**Issue by johnthagen** [axnsan12/drf-yasg#332]
Cross posted from https://github.com/Hipo/drf-extra-fields/issues/66

My model:

```py
class Image(models.Model):
      image=models.ImageField(upload_to='photos')
      name=models.CharField(max_length=40, unique=False)
```

My serializer:

```py
class imagesSerializer(serializers.ModelSerializer):
      image = Base64ImageField(required=True)
      class Meta:
         model = Image
         fields = ('id','name','image')
```

My swagger view:
![image](https://user-images.githubusercontent.com/10340167/54601717-1d21c880-4a16-11e9-9b76-3ed54c0b4011.png)

How can I make the `ImageField` as required in swagger POST request?

In the other thread we seem to have narrowed the problem down to `drf-yasg`, testing with `drf-yasg==1.13.0`.

**Comment by axnsan12** [comment]
A request body can only contain files when it has the multipart content type. You need to set the view parser_classes to `(MultipartParser,)` (no json).

**Comment by johnthagen** [comment]
@axnsan12 [`Base64ImageField`](https://github.com/Hipo/drf-extra-fields#base64imagefield) returns the contents of the field as base64 encoded ascii inline. It's used to make the API json-only so that multipart doesn't need to handled. This simplifies (in one degree) the overall API.

It sounds like `drf-yasg` assumes that all files must be sent multipart instead of inline?

This SO post explains the problem and `Base64ImageField` is one possible solution: https://stackoverflow.com/a/4083908/1398841

**Comment by axnsan12** [comment]
> It sounds like drf-yasg assumes that all files must be sent multipart instead of inline?

That is indeed the case. The field you linked breaks the contract of ImageField by making it accept a string instead of an UploadedFile.

You can work around this with a custom `FieldInspector` that returns an appropriate Schema object of type string, which should be placed before the FileFieldInspector.

**Comment by johnthagen** [comment]
@axnsan12 I'd like to try to fix this for `drf-extra-fields`. I'm not very familiar with custom `FieldInspector`'s. At a high level, what would need to be placed where? Would this be added to `Base64ImageField`? Are there any links to docs or guides that will help me learn what all the steps I need to take are and how to test?

Thanks!

**Comment by johnthagen** [comment]
PR with doc fix for this issue: #445
